src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,208 @@
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as FirebaseError from "./error";
+import * as logger from "./logger";
+import { pollOperation } from "./operation-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum AppPlatform {
+  IOS = "iOS",
+  ANDROID = "Android",
+  WEB = "Web",
+}
+
+export interface AppMetadata {
+  appId: string;
+  displayName: string;
+  appPlatform: AppPlatform;
+}
+
+export interface IosAppMetadata extends AppMetadata {
+  bundleId: string;
+  appPlatform: AppPlatform.IOS;
+}
+
+export interface AndroidAppMetadata extends AppMetadata {
+  packageName: string;
+  shaCertificates?: ShaCertificate[];
+  appPlatform: AppPlatform.ANDROID;
+}
+
+export interface WebAppMetadata extends AppMetadata {
+  appPlatform: AppPlatform.WEB;
+}
+
+export interface ShaCertificate {
+  certType: "SHA_1" | "SHA_256";
+  shaHash: string;
+}
+
+/**
+ * Send an API request to create a new Firebase iOS app and poll the LRO to get the new app
+ * information.
+ * @return {Promise} this function returns a promise that resolves to the new iOS app information
+ */
+export async function createIosApp(
+  projectId: string,
+  displayName: string,
+  bundleId: string
+): Promise<IosAppMetadata> {
+  const spinner = new OraWrapper(`Creating your ${clc.red("iOS")} app`);
```

Comment:
I think the CLI's convention is to only use red for errors. We could pick a different color, but we probably don't need coloring at all here.

Reply:
Removed the color